plugin/trino-hive/src/main/java/io/trino/plugin/hive/ViewReaderUtil.java

```diff
@@ -145,6 +147,11 @@ public static boolean isHiveOrPrestoView(String tableType)
         return tableType.equals(TableType.VIRTUAL_VIEW.name());
     }
 
+    public static boolean isTrinoMaterializedView(String tableType, Map<String, String> tableParameters)
+    {
+        return isHiveOrPrestoView(tableType) && isPrestoView(tableParameters) && tableParameters.containsKey(STORAGE_TABLE);
```

Comment:
I think `tableParameters.containsKey(STORAGE_TABLE)` should be replaced by `table.getParameters().get(TABLE_COMMENT).equalsIgnoreCase(ICEBERG_MATERIALIZED_VIEW_COMMENT)` here. Then any code using storage table from the table parameters should `verify(tableParameters.containsKey(STORAGE_TABLE))`